### PR TITLE
docs: various improvements to documentations

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] ~Addresses an existing issue: fixes #000~
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

changes that i missed in #4128

- properly use ✅ instead of ❌for correct examples
- validate some examples
- update remark tabs plugin
- add missing notes

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/unified-signatures.md [hunk 1]
```diff
@@ -6,26 +6,30 @@ Warns for any two overloads that could be unified into one by using a union or a
...
+### ✅ Correct
 
 ```ts
-function f(x: number | string): void;
+function x(x: number | string): void;
 ```
 
 ```ts
-function f(x?: ...number[]): void;
+function y(...x: number[]): void;
```
Comment (armano2) [inline packages/eslint-plugin/docs/rules/unified-signatures.md:32]: this example was incorrect

File: packages/eslint-plugin/docs/rules/unified-signatures.md [hunk 2]
```diff
@@ -6,26 +6,30 @@ Warns for any two overloads that could be unified into one by using a union or a
...
-function f(x: string): void;
+function x(x: number): void;
+function x(x: string): void;
 ```
 
 ```ts
-f(): void;
-f(...x: number[]): void;
+function y(): void;
+function y(...x: number[]): void;
```
Comment (armano2) [inline packages/eslint-plugin/docs/rules/unified-signatures.md:22]: missing `function`

File: packages/website/src/remark/tabs.js [hunk 3]
```diff
@@ -72,38 +68,45 @@ function findTabs(node, index, parent) {
...
 
-        return index + newChildren.length;
+            return index + newChildren.length;
+          }
+        }
       }
     });
 
-    if (foundTabs) {
+    if (foundTabs && !alreadyImported) {
```
Comment (armano2) [inline packages/website/src/remark/tabs.js:109]: changes done in https://github.com/mccleanp/remark-docusaurus-tabs/pull/2

this package is not yet released

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by bradzacher — APPROVED** [review]
we could probably add a test to ensure that the emojis are correct and help us validate doc structure

**Pull request merged**